coreneuron/io/output_spikes.cpp

```diff
@@ -268,8 +268,10 @@ void output_spikes_serial(const char* outpath) {
 
 void output_spikes(const char* outpath, const std::string& population_name) {
     // try to transfer spikes to NEURON. If successfull, don't write out.dat
-    if (all_spikes_return(spikevec_time, spikevec_gid))
+    if (all_spikes_return(spikevec_time, spikevec_gid)) {
+        clear_spike_vectors();
         return;
+    }
```

Comment:
Never fails to surprise me that something can be right in front of my nose and I can't see it til someone says, "see!"